### PR TITLE
TransactionController: providerConfig -> selectedNetworkClientId

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

The `providerConfig` state property is being removed from NetworkController. Currently this property is used in TransactionController to get the currently selected chain, but `selectedNetworkClientId` can be used instead. This commit makes that transition so that we can fully drop `providerConfig`.

## References

Progresses #4185.

## Changelog

(There should be no functional changes, so no changelog updates are needed.)

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/TransactionController.test.ts [hunk 1]
```diff
@@ -2015,10 +2009,13 @@ describe('TransactionController', () => {
...
 
         it('if unexpected status', async () => {
-          const { controller, messenger } = setupController();
-
-          jest.spyOn(messenger, 'call').mockImplementationOnce(() => {
-            throw new Error('Unknown problem');
+          const { controller } = setupController({
+            messengerOptions: {
+              addTransactionApprovalRequest: {
+                state: 'rejected',
```
Comment (MajorLift) [inline packages/transaction-controller/src/TransactionController.test.ts:2015]: nit:
```suggestion
                state: TransactionStatus.rejected,
```

File: packages/transaction-controller/src/TransactionController.test.ts [hunk 2]
```diff
@@ -2033,10 +2030,13 @@ describe('TransactionController', () => {
...
 
         it('if unrecognised error', async () => {
-          const { controller, messenger } = setupController();
-
-          jest.spyOn(messenger, 'call').mockImplementationOnce(() => {
-            throw new Error('TestError');
+          const { controller } = setupController({
+            messengerOptions: {
+              addTransactionApprovalRequest: {
+                state: 'rejected',
```
Comment (MajorLift) [inline packages/transaction-controller/src/TransactionController.test.ts:2036]: nit:
```suggestion
                state: TransactionStatus.rejected,
```

**Review by MajorLift — COMMENTED** [review]
LGTM! Just need the merge conflict to be resolved.

File: packages/transaction-controller/src/TransactionController.test.ts (hunk 1 above)
Comment (mcmire) [reply packages/transaction-controller/src/TransactionController.test.ts:2015]: Good point.

File: packages/transaction-controller/src/TransactionController.test.ts (hunk 2 above)
Comment (mcmire) [reply packages/transaction-controller/src/TransactionController.test.ts:2036]: Makes sense.

**Review by cryptodev-2s — APPROVED** [review]
LGTM!

**Pull request merged**